### PR TITLE
Documentation: add public key link to release doc

### PR DESCRIPTION
**Pull request by gyuho**
/cc @xiang90 @philips

### CODE REVIEW

File: Documentation/dev/release.md [hunk 1]
```diff
@@ -70,6 +70,9 @@ for i in etcd-*{.zip,.tar.gz}; do gpg2 --default-key $SUBKEYID --output ${i}.asc
 for i in etcd-*{.zip,.tar.gz}; do gpg2 --verify ${i}.asc ${i}; done
 ```
 
+The public key used to sign etcd binaries can be found at [coreos.com/security](https://coreos.com/security/app-signing-key)
```
Comment (gyuho) [inline Documentation/dev/release.md]: Did clarify about which key we are using for signing etcd binaries. PTAL. Thanks.

Comment (xiang90) [inline Documentation/dev/release.md]: The signing GPG key can be found at ...

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
thanks.

**Pull request merged**